controlplane/controllers/scale.go

```diff
@@ -159,6 +160,27 @@ func (r *CK8sControlPlaneReconciler) scaleDownControlPlane(
 	}
 	**/
 
+	clusterObjectKey := util.ObjectKey(cluster)
+	workloadCluster, err := r.managementCluster.GetWorkloadCluster(ctx, clusterObjectKey)
+	if err != nil {
+		logger.Error(err, "failed to create client to workload cluster")
+		return ctrl.Result{}, errors.Wrapf(err, "failed to create client to workload cluster")
+	}
+
+	authToken, err := token.Lookup(ctx, r.Client, clusterObjectKey)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to lookup auth token: %w", err)
+	}
+
+	if authToken == nil {
+		return ctrl.Result{}, fmt.Errorf("auth token not yet generated")
+	}
```

Comment:
The token lookup feels a bit out of place here, maybe it could be included in the GetWorkloadCluster, or perhaps the `RemoveMachineFromCluster` call. But that's ok, let's sleep on this and see how the code evolves before any changes.

Reply:
Moved it to the GetWorkloadCluster call, feels quite right there.